src/pcd/mod.rs

```diff
@@ -8,41 +8,30 @@ pub mod receipt;
 #[cfg(test)]
 mod tests;
 use anyhow::anyhow;
-use nova::provider::PallasEngine;
 
 use crate::{
-  circuits::{supernova::etable_rom::wasm_nivc, verify::verify_receipts},
+  circuits::verify::verify_receipts,
   errors::ProvingError,
+  provider::E,
+  run::batched,
   traits::wasm::ZKWASMContext,
-  utils::nivc::build_rom,
+  wasm::{args::WASMArgs, ctx::wasi::WasiWASMCtx},
 };
 
 use receipt::Receipt;
 
-// TODO: improve time complexity
 /// Prove WASM shard. (Top-Down approach)
 /// Returns a receipt which can be used to verify the proof & show that related shards are
 /// connected.
 ///
 /// `start` and `end` are parameters to specify the shards range of opcodes to prove.
-pub fn prove_shard(
-  wasm_ctx: &mut impl ZKWASMContext,
-  should_stop: impl Fn() -> bool,
```

Comment:
Not having ability to interrupt the proving process means that if a job needs to be canceled (it was finalized by some other ON), this prover will waste its time proving a shard no one's interested in. If we don't need this feature anymore because the batched mode is that much faster, I'm happy to see this gone.

If we still want to have that ability, maybe there's a "zkEngine native" way of interrupting the proving process, instead of this ugly hack of passing `impl Fn()` around.

Reply:
I’m not entirely sure, but let’s remove it temporarily, if that's okay. We can revisit it later since the entire sharding process in zkEngine is due for a revamp